File: BooleanLogicAssignment.cs
using System;

namespace BooleanLogicAssignment 
class Program
{
    static void Main()
    {
        // Ask the user for age
        Console.Write("What is your age? ");
        int age = Convert.ToInt32(Console.ReadLine());

        // Ask about DUI
        Console.Write("Have you ever had a DUI? (yes or no) ");
        string duiInput = Console.ReadLine().ToLower();
        bool hasDui = duiInput == "yes";

        // Ask about speeding tickets
        Console.Write("How many speeding tickets do you have? ");
        int speedingTickets = Convert.ToInt32(Console.ReadLine());

        // Check qualification rules
        bool isEligible = age > 15 && !hasDui && speedingTickets <= 3;

        // Print result
        Console.WriteLine(isEligible);
    }
}

File: BooleanLogicAssignment.py
def check_insurance_eligibility():
    # Get user input
    age = int(input("What is your age? "))
    has_dui = input("Have you ever had a DUI? (yes or no) ").strip().lower()
    speeding_tickets = int(input("How many speeding tickets do you have? "))

    # Convert DUI answer to boolean
    dui = has_dui == "yes"

    # Apply qualification rules
    is_eligible = (age > 15) and (not dui) and (speeding_tickets <= 3)

    # Print the result
    print(is_eligible)

# Run the function
check_insurance_eligibility()
